File: 백준/Gold/2467. 용액/용액.py
import sys
input = sys.stdin.readline

n = int(input())
liquid = list(map(int, input().split()))

left = 0
right = n-1
answer = []
min_value = sys.maxsize
while left < right:
    value = liquid[right] + liquid[left]
    if abs(value) < min_value:
        min_value = abs(value)
        answer = [liquid[left], liquid[right]]
    if value > 0:
        right -= 1
    elif value < 0:
        left += 1
    else:
        break

print(*answer)

File: 백준/Gold/2467. 용액/용액.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const N = Number(input[0])
const arr = input[1].split(' ').map(Number)
arr.sort((a, b) => a - b)

const value = []
let left = 0
let right = N - 1
while(left < right) {
  const diff = arr[right] + arr[left]
  value.push({ diff: Math.abs(diff), left, right })

  if (diff < 0) {
    left++
  } else if (0 < diff) {
    right--
  } else {
    break
  }
}

value.sort((a, b) => a.diff - b.diff)
console.log(arr[value[0].left], arr[value[0].right])